src/main/java/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java

```diff
@@ -354,22 +357,25 @@ public void pushRepoToTag(String repoModuleName, RepoTag tag, String userId)
             throw new NarrativeMethodStoreException("Unsupported tag: " + tag);
         }
         // race condition here
-        data.update(new BasicDBObject(FIELD_RI_MODULE_NAME, repoModuleName),
-                new BasicDBObject("$set", info));
-        final DBCollection data2 = db.getCollection(TABLE_REPO_HISTORY);
-        final RepoHistory hist = toObject(data2.findOne(
-                new BasicDBObject(FIELD_RH_MODULE_NAME, repoModuleName)
-                        .append(FIELD_RH_VERSION, changedVer)), RepoHistory.class);
+        data.updateOne(Filters.eq(FIELD_RI_MODULE_NAME, repoModuleName), new Document("$set", info));
+
+        final MongoCollection<Document> data2 = db.getCollection(TABLE_REPO_HISTORY);
+
+        final RepoHistory hist = toObject(data2.find(Filters.and(
+                Filters.eq(FIELD_RH_MODULE_NAME, repoModuleName),
+                Filters.eq(FIELD_RH_VERSION, changedVer))).first(), RepoHistory.class);
+
         hist.repo_data.repackForMongoDB();
         if (tag.equals(RepoTag.beta)) {
             hist.is_beta = 1L;
         } else {
             hist.is_release = 1L;
         }
-        // race condition here
```

Comment:
I think this comment is still relevant

Reply:
same as https://github.com/kbase/narrative_method_store/pull/99#discussion_r1815616417?

Reply:
yeah, looks like it

Reply:
👍